src/sentry/issues/producer.py

```diff
@@ -26,25 +31,32 @@ def _get_occurrence_producer() -> KafkaProducer:
 
 
 def produce_occurrence_to_kafka(
-    occurrence: IssueOccurrence, event_data: Optional[Dict[str, Any]] = None
+    payload_type: PayloadType | None = PayloadType.OCCURRENCE,
+    occurrence: IssueOccurrence | None = None,
+    event_data: Optional[Dict[str, Any]] = None,
 ) -> None:
-    if event_data and occurrence.event_id != event_data["event_id"]:
-        raise ValueError("Event id on occurrence and event_data must be the same")
-    if settings.SENTRY_EVENTSTREAM != "sentry.eventstream.kafka.KafkaEventStream":
-        # If we're not running Kafka then we're just in dev. Skip producing to Kafka and just
-        # write to the issue platform directly
-        from sentry.issues.occurrence_consumer import (
-            lookup_event_and_process_issue_occurrence,
-            process_event_and_issue_occurrence,
-        )
-
+    if payload_type == PayloadType.OCCURRENCE:
+        if not occurrence:
+            raise ValueError("Occurrence must be provided")
+        if event_data and occurrence.event_id != event_data["event_id"]:
+            raise ValueError("Event id on occurrence and event_data must be the same")
+        if settings.SENTRY_EVENTSTREAM != "sentry.eventstream.kafka.KafkaEventStream":
+            # If we're not running Kafka then we're just in dev. Skip producing to Kafka and just
+            # write to the issue platform directly
+            from sentry.issues.occurrence_consumer import (
+                lookup_event_and_process_issue_occurrence,
+                process_event_and_issue_occurrence,
+            )
+
+            if event_data:
+                process_event_and_issue_occurrence(occurrence.to_dict(), event_data)
+            else:
+                lookup_event_and_process_issue_occurrence(occurrence.to_dict())
+            return
+        payload_data = cast(MutableMapping[str, Any], occurrence.to_dict())
```

Comment:
why do we have to cast here? should be avoided if possible

Reply:
Not sure, we're just indenting the code here but can look into cleaning it up later.